Wrong result on slop query

With a slop set to 0, this query shouldn't return any document

```
127.0.0.1:6379> FT.CREATE idx ON HASH SCHEMA label TEXT
OK
127.0.0.1:6379> HSET doc:1 label "Hello the world"
(integer) 1
127.0.0.1:6379> FT.SEARCH idx '(Hello world) => {$slop: 0}'
1) (integer) 1
``` 

```
======================================================================
FAIL: testClient (__main__.RedisSearchTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/circleci/repo/test/test.py", line 169, in testClient
    self.assertEqual(3,client.search(Query('henry king').slop(0).in_order()).total)
AssertionError: 3 != 52
```
https://app.circleci.com/pipelines/github/RediSearch/redisearch-py/654/workflows/7dcd6c25-6c3e-44ec-8eb5-cef51f72eda0/jobs/1325

--- a/src/index.c
+++ b/src/index.c
@@ -781,7 +781,9 @@ IndexIterator *NewIntersecIterator(IndexIterator **its_, size_t num, DocTable *d
   ctx->num = num;
 
   // Sort children iterators from low count to high count which reduces the number of iterations.
-  qsort(ctx->its, ctx->num, sizeof(*ctx->its), (CompareFunc)cmpIter);
+  if (!ctx->inOrder) {
+    qsort(ctx->its, ctx->num, sizeof(*ctx->its), (CompareFunc)cmpIter);
+  }
 
   // bind the iterator calls
   IndexIterator *it = &ctx->base;

--- a/tests/pytests/test.py
+++ b/tests/pytests/test.py
@@ -975,16 +975,22 @@ def testSlopInOrder(env):
     env.assertEqual(0, r.execute_command(
         'ft.search', 'idx', 't1 t2 t3 t4', 'inorder')[0])
 
-    # test with qsort optimization on intersect iterator
-    # increase the count for t1 inverted index so optimization is applied
-    env.assertOk(r.execute_command('ft.add', 'idx', 'doc6', 1, 'fields',
-                                    'title', 't1'))
-    # add a field with different order of `t1, t2` 
-    env.assertOk(r.execute_command('ft.add', 'idx', 'doc5', 1, 'fields',
-                                    'title', 't2 t1'))
 
-    # before fix, both queries returned `doc5`
-    env.assertEqual([1L, 'doc5', ['title', 't2 t1']], r.execute_command(
+def testSlopInOrderIssue1986(env):
+    r = env
+    # test with qsort optimization on intersect iterator
+    env.assertOk(r.execute_command(
+        'ft.create', 'idx', 'ON', 'HASH', 'schema', 'title', 'text'))
+
+    env.assertOk(r.execute_command('ft.add', 'idx', 'doc1', 1, 'fields',
+                                    'title', 't1 t2'))
+    env.assertOk(r.execute_command('ft.add', 'idx', 'doc2', 1, 'fields',
+                                    'title', 't2 t1'))
+    env.assertOk(r.execute_command('ft.add', 'idx', 'doc3', 1, 'fields',
+                                    'title', 't1'))
+
+    # before fix, both queries returned `doc2`
+    env.assertEqual([1L, 'doc2', ['title', 't2 t1']], r.execute_command(
         'ft.search', 'idx', 't2 t1', 'slop', '0', 'inorder'))
     env.assertEqual([1L, 'doc1', ['title', 't1 t2']], r.execute_command(
         'ft.search', 'idx', 't1 t2', 'slop', '0', 'inorder'))

--- a/tests/pytests/test.py
+++ b/tests/pytests/test.py
@@ -975,6 +975,20 @@ def testSlopInOrder(env):
     env.assertEqual(0, r.execute_command(
         'ft.search', 'idx', 't1 t2 t3 t4', 'inorder')[0])
 
+    # test with qsort optimization on intersect iterator
+    # increase the count for t1 inverted index so optimization is applied
+    env.assertOk(r.execute_command('ft.add', 'idx', 'doc6', 1, 'fields',
+                                    'title', 't1'))
+    # add a field with different order of `t1, t2` 
+    env.assertOk(r.execute_command('ft.add', 'idx', 'doc5', 1, 'fields',
+                                    'title', 't2 t1'))
+
+    # before fix, both queries returned `doc5`
+    env.assertEqual([1L, 'doc5', ['title', 't2 t1']], r.execute_command(
+        'ft.search', 'idx', 't2 t1', 'slop', '0', 'inorder'))
+    env.assertEqual([1L, 'doc1', ['title', 't1 t2']], r.execute_command(
+        'ft.search', 'idx', 't1 t2', 'slop', '0', 'inorder'))
+
 def testExact(env):
     r = env
     env.assertOk(r.execute_command(

